src/phoenix/trace/span_evaluations.py

```diff
@@ -152,6 +157,24 @@ def __init_subclass__(
             tuple(sorted(prod)) for prod in product(*cls.index_names.keys())
         )
 
+    def to_parquet(self, path: Union[str, Path]) -> None:
+        table = Table.from_pandas(self.dataframe)
+        table = table.replace_schema_metadata(
+            {
+                **table.schema.metadata,
+                b"openinference": json.dumps({"eval_name": self.eval_name}).encode("utf-8"),
+            }
+        )
+        parquet.write_table(table, path)
+
+    @classmethod
+    def from_parquet(cls: EvaluationsType, path: Union[str, Path]) -> EvaluationsType:
+        table = parquet.read_table(path)
```

Comment:
Maybe we should read and validate the schema first, before reading the whole table.

Here's an exaggerated timing difference.

<img width="389" alt="Screenshot 2023-12-28 at 12 11 38 PM" src="https://github.com/Arize-ai/phoenix/assets/80478925/341f7be6-4e10-419f-b05a-19194c92d0b7">

Reply:
Good call on validation of the schema.